src/megachatapi_impl.cpp

```diff
@@ -5282,10 +5328,29 @@ MegaChatSessionHandler::~MegaChatSessionHandler()
 
 void MegaChatSessionHandler::onSessStateChange(uint8_t newState)
 {
-    if (newState == rtcModule::ISession::kStateInProgress)
+    MegaChatCallPrivate* chatCall = callHandler->getMegaChatCall();
+
+    assert(chatCall != NULL);
+    if (chatCall != NULL)
     {
-        MegaChatCallPrivate* chatCall = callHandler->getMegaChatCall();
-        chatCall->setRemoteAudioVideoFlags(session->receivedAv());
+        if (newState == rtcModule::ISession::kStateInProgress)
+        {
+            chatCall->setRemoteAudioVideoFlags(session->receivedAv());
+
+            if (chatCall->getRemoteStatus() == rtcModule::ICall::kStateRingIn)
+            {
+                chatCall->setRemoteStatus(rtcModule::ICall::kStateInProgress);
+            }
+        }
+        else if (newState == rtcModule::ISession::kStateTerminating)
+        {
+            chatCall->setRemoteStatus(rtcModule::ICall::kStateTerminating);
+        }
+        else if (newState == rtcModule::ISession::kStateDestroyed)
+        {
+            chatCall->setRemoteStatus(rtcModule::ICall::kStateDestroyed);
+        }
+
```

Comment:
I'd say we should simply `chatCall->setRemoteStatus(newState)` (apart from set the AV flags upon transition to `inProgress`).
If the new status is none of the expected ones, assert plus log message.

Reply:
According to our conversation, the session state can be other than in-progress, terminating or destroyed, so we need to trigger a `onChatCallUpdate()` only for those 3 cases that are of some interest to the apps.